lib/dest/writeContents/index.js

```diff
@@ -44,12 +44,12 @@ function writeContents(writePath, file, cb) {
       if (err) {
         return complete(err);
       }
-      // octal 7777 = decimal 4095
-      var currentMode = (st.mode & 4095);
-      if (currentMode === file.stat.mode) {
+      var currentMode = (st.mode & parseInt('0777', 8));
+      var expectedMode = (file.stat.mode & parseInt('0777', 8));
+      if (currentMode === expectedMode) {
```

Comment:
parseInt is just there for readability, parseInt('07777', 8) would give 4095.  
I changed it to 0777 to remove special bits, for example : 
- given a folder with sticky bit and setuid (5722)
- if you then make a file with mode 644, it will become 5644
- as of master, `currentMode === file.stat.mode` evaluates to false
- vinyl-fs then tries to chmod the file, which fails when you are not the owner

My opinion is to let the filesystem handle the special bits and just deal with the actual permissions